File: exemplos/aula_250617/example2.java
import java.util.Scanner;

class RaizQuadradaException extends Exception {
    public RaizQuadradaException() {
        super("Não é possível calcular a raiz quadrada de um número negativo!");
    }
    
}

class Utils{
    public static double raizQuadrada(double numero) throws RaizQuadradaException {
        if (numero < 0) {
            throw new RaizQuadradaException();
        }
        return Math.sqrt(numero);
    }
}

class Example2{
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        try {
            double numero = scanner.nextDouble();
            double resultado = Utils.raizQuadrada(numero);
            System.out.println("A raiz quadrada de " + numero + " é: " + resultado);
        } catch (RaizQuadradaException e) {
            System.out.println("Erro: " + e.getMessage());
        } catch (Exception e) {
            System.out.println("Ocorreu um erro inesperado: " + e.getMessage());
        } finally {
            scanner.close();
            System.out.println("Fim da operação! 🎉");
        }
    }
}

File: exemplos/aula_250617/example2.py
class RaizQuadradaError(Exception):
    def __init__(self):
        super().__init__("Erro ao calcular a raiz quadrada")

class Utils:
    @staticmethod
    def calcular_raiz_quadrada(numero):
        if numero < 0:
            raise RaizQuadradaError()
        return numero ** 0.5

try:
    numero = float(input("Digite um número para calcular a raiz quadrada: "))
    resultado = Utils.calcular_raiz_quadrada(numero)
    print(f"A raiz quadrada de {numero} é {resultado}")
except RaizQuadradaError as e:
    print(f"Erro: {e} 🚫")
except Exception as e:
    print(f"Ocorreu um erro inesperado: {e} ❗")
